src/components/RangeSelector.js

```diff
@@ -92,6 +92,68 @@ class RangeSelector extends React.Component {
     });
   }
 
+  _handleClickStartOnTrack (e) {
+    const clientX = e.touches ? e.touches[0].clientX : e.clientX;
+
+    const newPixels = clientX - ReactDOM.findDOMNode(this.refs.rangeSelector).getBoundingClientRect().left;
+
+    //click point is less than lower selector
+    if (newPixels < this.state.lowerPixels) {
+      this.setState({
+        dragging: 'Lower'
+      });
+    }
+
+    //click point is higher than upper selector
+    if (newPixels > this.state.upperPixels) {
+      this.setState({
+        dragging: 'Upper'
+      });
+    }
+
+    //click point is higher than lower selector && also lower than midway point between selectors
+    if (newPixels > this.state.lowerPixels && newPixels < (this.state.lowerPixels + (this.state.upperPixels - this.state.lowerPixels) / 2)) {
+      this.setState({
+        dragging: 'Lower'
+      });
+    }
+
+    //click point is lower than upper selector && also greater than midway point between selectors
+    if (newPixels < this.state.upperPixels && newPixels > (this.state.upperPixels - (this.state.upperPixels - this.state.lowerPixels) / 2)) {
+      this.setState({
+        dragging: 'Upper'
+      });
+    }
+  }
+
+  _handleClickOnTrack (e) {
+    if (this.state.dragging) {
+      const clientX = e.touches ? e.touches[0].clientX : e.clientX;
+      const pixelInterval = this.props.interval * this.state.width / this.state.range;
+      const newState = {
+        selectedLabel: null
+      };
+
+      let newPixels = clientX - ReactDOM.findDOMNode(this.refs.rangeSelector).getBoundingClientRect().left;
+
+      //make sure we snap to our interval
+      newPixels = Math.round(newPixels / pixelInterval) * pixelInterval;
+
+      //convert our pixels to a 0-based scale
+      const newPosition = (newPixels * this.state.range / this.state.width) + this.props.lowerBound;
+
+      //covert our 0-based value to actual value
+      const newValue = Math.round(newPosition / this.props.interval) * this.props.interval;
+
+      newState[this.state.dragging.toLowerCase() + 'Pixels'] = newPixels;
+      newState[this.state.dragging.toLowerCase() + 'Value'] = newValue;
+
+      this.setState(newState);
+
+      e.preventDefault();
```

Comment:
So what's happening is the event is bubbling up, so the track `onMouseUp` event fires, then the `mx-rangeselector-range` div `onMouseUp` event fires. The result is we're setting state twice for one event (which we want to avoid). One way to avoid this would be to eliminate this method all together and rely on the `mx-rangeselector-range` div `onMouseUp` event to handle this logic. If you do this, then you'll want to set some type of state attribute in `_handleClickStartOnTrack` to indicate a track click happened.

The `onLowerDragStop` and `onUpperDragStop` methods are passed to this component through it's props. So they are defined by the parent component. When called, the parent component will be "notified" of the toggle value change. @cerinman can you think of a better way to explain this?